pkg/controller/argocd/role.go

```diff
@@ -107,20 +107,30 @@ func (r *ReconcileArgoCD) reconcileRole(name string, policyRules []v1.PolicyRule
 }
 
 func (r *ReconcileArgoCD) reconcileClusterRole(name string, policyRules []v1.PolicyRule, cr *argoprojv1a1.ArgoCD) (*v1.ClusterRole, error) {
-
 	clusterRole := newClusterRole(name, policyRules, cr)
-	err := r.client.Get(context.TODO(), types.NamespacedName{Name: clusterRole.Name}, clusterRole)
+	clusterRole.Rules = policyRules
```

Comment:
this line is redundant and not needed.

Reply:
`ReconcileClusterRole` is a generic function. I know right now the way we are calling it seems redundant but in the future, there can be a possibility where we need to create ClusterRoles and reconcile them. So it is for that.

Reply:
I think we should remove it if it is redundant and not needed.

Reply:
I think it's still needed because the definition of the function is to append the policy rules to the cluster role provided. Right now we are not passing any rules and only rely on the rules provided by the reconciler hook. 
But according to the definition of the function `reconcileClusterRole`, we should try and append the policy rules provided, irrespective of what is being provided.

Reply:
The line above it (L110) creates the `clusterRole` object with `policyRules` as an argument.  So, the `Rules` field of `clusterRole` already has `policyRules` assigned.  If you remove the line `clusterRole.Rules = policyRules` what problem does it cause?    I am missing something?

Reply:
ohhh sorry, I misunderstood your comment, Yes, that makes total sense, this line is redundant and I should remove it. Thanks !

Reply:
Thanks for the comment, fixed it !!